tests/Loader/LoaderIntegrationTest.php

```diff
@@ -581,6 +583,43 @@ public function testLoadSelfReferencedFixtures()
         $this->assertCount(2, $objects);
     }
 
+    public function testLoadRangeReference()
+    {
+        $data = [
+            User::class => [
+                'user0' => [
+                    'name' => '<username()>',
+                ],
+                'user1' => [
+                    'name' => '<username()>',
+                ],
+            ],
+            UserDetail::class => [
+                'userdetail_{@user*}' => [
+                    'email' => '<email()>',
+                    'user'  => '<current()>',
+                ],
+                'userdetail_single_{@user1}' => [
+                    'email' => '<email()>',
+                    'user'  => '<current()>',
+                ],
+            ],
+        ];
+
+        $set = $this->loader->loadData($data);
+
+        $objects = $set->getObjects();
+        $this->assertCount(5, $objects);
+
+        $this->assertArrayHasKey('userdetail_user0', $objects);
+        $this->assertArrayHasKey('userdetail_user1', $objects);
+        $this->assertArrayHasKey('userdetail_single_user1', $objects);
+
+        $this->assertInstanceOf(User::class, $objects['userdetail_user0']->getUser());
```

Comment:
maybe more than `assertInstanceOf()` we could do a `assertSame()` as the instance should be the same